File: frontend/src/types/note.ts
export interface Note {
  id: string;
  title: string;
  content: string;
  created_at: string;
}

export interface NoteCreate {
  title: string;
  content: string;
}

export type RequestStatus = 'idle' | 'loading' | 'succeeded' | 'failed';

File: backend/src/app/models/note.py
"""Pydantic models for the Note domain."""

from datetime import datetime

from pydantic import BaseModel, Field


class NoteCreate(BaseModel):
    """Schema for creating a new note."""

    title: str = Field(..., min_length=1, max_length=200)
    content: str = Field(..., min_length=1)


class NoteResponse(BaseModel):
    """Schema for note API responses."""

    id: str
    title: str
    content: str
    created_at: datetime
